sre/roles/faults/tasks/inject_low_resource_limits.yaml

```diff
@@ -0,0 +1,66 @@
+---
+- name: Retrieve workload information for target services
+  kubernetes.core.k8s_info:
+    api_version: apps/v1
+    kind: "{{ workload.kind }}"
+    kubeconfig: "{{ faults_cluster.kubeconfig }}"
+    name: "{{ workload.name }}"
+    namespace: "{{ spec.namespace.name }}"
+  register: faults_workloads_info
+  loop: "{{ spec.workloads }}"
+  loop_control:
+    label: "{{ workload.kind | lower }}/{{ workload.name }}"
+    loop_var: workload
+  when:
+    - workload.name in ['valkey', 'postgresql']  # Only target specific services
+
+- name: Patch workloads with restrictive resource limits
+  kubernetes.core.k8s:
+    kubeconfig: "{{ faults_cluster.kubeconfig }}"
+    state: patched
+    api_version: "{{ result.resources[0].apiVersion }}"
+    kind: "{{ result.resources[0].kind }}"
+    name: "{{ result.resources[0].metadata.name }}"
+    namespace: "{{ result.resources[0].metadata.namespace }}"
+    definition:
+      spec:
+        template:
+          spec:
+            containers:
+            - name: "{{ result.resources[0].spec.template.spec.containers[0].name }}"
```

Comment:
Maybe do something like [this](https://github.com/itbench-hub/ITBench-Scenarios/blob/aa3cb532dbccfa63239acc92dec5151e5f052785/sre/roles/faults/tasks/inject_custom_invalid_command.yaml) instead. Should the container being targeted not be the first in the index, this will overwrite it (and the entire array itself), which is not desired.